src/signer.ts

```diff
@@ -50,6 +54,11 @@ export interface GetSignedUrlConfigInternal {
   contentType?: string;
   bucket: string;
   file?: string;
+  /**
+   * @example
+   * 'https://iamcredentials.googleapis.com/v1/projects/-/serviceAccounts/'
+   */
+  signingEndpoint?: string;
```

Comment:
Interesting addition! What use-case do you envision here?

Reply:
Customers can use this to set customize the signed URL endpoint for the signed URL V2 & V4 flows, which is passed to auth via `auth.sign`/`authClient.sign`

Reply:
I guess I'm confused by the example - it seems like you are passing in what to call to create the signURL, not where the signed URL endpoint will point to.

Reply:
Yea, the `cname` is what the url will point to

Reply:
`cname` works for all styles, not just Bucketbound?

Reply:
Yep, see:

https://github.com/googleapis/nodejs-storage/blob/0d3e2111df5c3f7570c07b986c7b4f6d560e888a/src/signer.ts#L251-L254

Reply:
I guess I'm confused here...
`cname` is a specific concept that is tied to [bucket bound hostnames]( https://cloud.google.com/storage/docs/request-endpoints#cname), and while it does seem to allow arbitrary hosts, the naming is confusing. Ultimately it is not a requirement that a hostname can be set directly when signing, so I think this is okay though.

What I think this PR misses addressing is the FR of when a custom endpoint is set on the client, Sign URLs (v2 and v4) and Post Policies automatically use that endpoint. I think if it did that, specifically setting the URL in the universe domain case in the quoted code may be unnecessary (I guess it depends on how you implement it).

Reply:
I figured out the gap in understanding - should the `hostname` be included in the the signature?